app/controllers/homepage_controller.rb

```diff
@@ -177,8 +162,42 @@ def find_listings(params, listings_per_page, filter_params, includes, location_s
       price_min: params[:price_min],
       price_max: params[:price_max],
       locale: I18n.locale,
-      include_closed: false
-    }.merge(search_extra)
+      include_closed: false,
+      sort: nil
+    }
+
+
+    # location search params
+    distance = params[:distance_max].to_f
+    distance_system = marketplace_configuration ? marketplace_configuration[:distance_unit] : nil
+    distance_unit = (location_search_in_use && distance_system == :metric) ? :km : :miles
+    limit_search_distance = marketplace_configuration ? marketplace_configuration[:limit_search_distance] : true
+    distance_limit = [distance, APP_CONFIG[:external_search_distance_limit_min]].max if limit_search_distance
+
+    if @view_type != 'map' && params[:lc]
```

Comment:
location_search_in_use might be better here since it's already available (its a check for params[:lc] + search mode (:location && :keyword_and_location)